test/cql-pytest/conftest.py

```diff
@@ -221,3 +221,14 @@ def temp_workdir():
     """ Creates a temporary work directory, for the scope of a single test. """
     with tempfile.TemporaryDirectory() as workdir:
         yield workdir
+
+@pytest.fixture(scope="function")
+def compact_storage(cql):
+    try:
+        with config_value_context(cql, 'enable_create_table_with_compact_storage', 'true') as ctx:
```

Comment:
This is a nice trick, but it adds a performance penalty (before and after every test you change the configuration) and complexity, and you actually missed one use in test_empty.py (just a single test function).

But I'm curious why you decided to do this instead of just enabliing the `enable_create_table_with_compact_storage` option in test/cql-pytest/run.py. That's what I would have done - and we already do in other places where we need to enable experimental behavior (e.g., user-defined functions).
I see this is exactly what you did for the topology tests (you added the option in test/pylib/scylla_cluster.py) and could have done the same thing for cql-pytest.

Reply:
> This is a nice trick, but it adds a performance penalty (before and after every test you change the configuration) and complexity, and you actually missed one use in test_empty.py (just a single test function).

Ah, looks like `test_insert_empty_string_key_compact` is marked as `xfail` so I missed it.

About the performance penalty I doubt if it's measurable.
I do prefer the default value in testing to reflect the default in production (i.e. `enable_create_table_with_compact_storage=false`) to make it harder to add new tests that might rely on it - as we often copy-paste existing tests cases when creating new ones.

> 
> But I'm curious why you decided to do this instead of just enabliing the `enable_create_table_with_compact_storage` option in test/cql-pytest/run.py. That's what I would have done - and we already do in other places where we need to enable experimental behavior (e.g., user-defined functions). I see this is exactly what you did for the topology tests (you added the option in test/pylib/scylla_cluster.py) and could have done the same thing for cql-pytest.

The change in `test/pylib/scylla_cluster.py` is needed for the cql tests, not for topology, as I found no wrapper around `cql/cdc_compact_storage_test` for which I could use a fixture to enable compact storage.

Reply:
As for the performance penalty, I changed the scope of the fixture from `function` to `session` in complete test files testing compact storage in v3.
And also added the `compact_storage` fixture to test_empty.py `test_insert_empty_string_key_compact` and `test_insert_empty_string_compound_clustering_key_compact`.

Reply:
I think this change is not a good direction because it relies of pytest not reordering the run order of tests - which you shouldn't rely on.